src/infrastructure/markup/blockrule/PhutilRemarkupCodeBlockRule.php

```diff
@@ -44,7 +44,14 @@ public function getMatchingLineCount(array $lines, $cursor) {
   }
 
   public function markupText($text, $children) {
+    $default_lang = null;
+
     if (preg_match('/^\s*```/', $text)) {
+      // Look for GFM-style ```langname blocks and handle them
+      if (preg_match('/\A\s*``` *([a-z]+)\n/', $text, $matches)) {
```

Comment:
this code uses ^ elsewhere rather than \A.

Reply:
I don't think it makes a difference unless PCRE_MULTILINE is set; I can change it if you prefer